### DISCUSSION TITLE
Evolutions suggestions about display and dashboard

### DISCUSSION THREAD

**Discussion by magicgg91** [umami-software/umami#609] [Ideas]
Hello,

I would suggest 3-4 evolutions that could be helpful in navigating/analyzing :
- Add keyboard shorcuts, to quickly display datas (d for today, w this week, m this month, y this year, c for custom) and left/right arrows to navigate between days/months
-  If we display a week or a month, a click on barchart of a particular day can zoom in 24h view for this day.
- Display % of evolution for pageviews/visitors/bounce rate/duration in comparison of previous period. Very helpful to not do it manually to see by how many our website growth and what is the trend.
- As a click on url displays datas only for this URL, a click on a referrer can display only datas for it, with details of urls reached with it.

Thanks for your soft and your work, it's very appreciated.

**Comment by mikecao** [comment]
1. Keyboard shortcuts could be possible
2. That is possible too
3. This will require twice the number of queries (current and previous time period) and would affect performance greatly.
4. Click on the referrer already takes you to the referrer url. May need to redesign how filtering is done.

**Comment by magicgg91** [comment]
1. & 2. Ok cool to hear that
3. I understand and agree that it shouldn't impact performances
4. Yes, it takes you to the referrer url, but in my suggestion, it shouldn't and would only apply a filter with the referrer url selected. Same behaviour as pages.

**Comment by magicgg91** [comment]
Point 3 is similar to https://github.com/mikecao/umami/issues/600

**Comment by mikecao** [comment]
I'm not going to get rid of the referrer link because it is useful. Will need to find a way to support both.

**Reply by magicgg91** [reply]
I agree that link is useful.

A solution can be to have referrer name with same behavior as page name, to apply filter. And to append a small icon for link to referrer. 
An example :

![referrer](https://user-images.githubusercontent.com/16479933/114469405-60e35e00-9bed-11eb-84cf-58f487226fa1.PNG)
